Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/view/CountryList.kt

```diff
@@ -2,47 +2,98 @@ package de.rki.coronawarnapp.ui.view
 
 import android.content.Context
 import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
 import android.widget.ImageView
 import android.widget.LinearLayout
-import android.widget.TextView
+import androidx.core.content.withStyledAttributes
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.RecyclerView
 import de.rki.coronawarnapp.R
 import de.rki.coronawarnapp.ui.Country
+import kotlinx.android.synthetic.main.view_country_list_entry_flag_container.view.*
 import java.text.Collator
 
 class CountryList(context: Context, attrs: AttributeSet) :
     LinearLayout(context, attrs) {
-
+    private var attributeSet = attrs
+    private var view: View? = null
     private var _list: List<Country>? = null
     var list: List<Country>?
         get() = _list
         set(value) {
             _list = value
-            buildList()
+            buildAllLists()
         }
 
     init {
-        orientation = VERTICAL
+        orientation = HORIZONTAL
     }
 
     /**
      * Cleans the view and rebuilds the list of countries. Presets already selected countries
      */
-    private fun buildList() {
-        this.removeAllViews()
+    private fun buildAllLists() {
         list
             ?.map { country ->
                 context.getString(country.labelRes) to country.iconRes
             }
             ?.sortedWith { a, b ->
                 Collator.getInstance().compare(a.first, b.first)
             }
-            ?.forEachIndexed { index, (label, iconRes) ->
-                inflate(context, R.layout.view_country_list_entry, this)
-                val child = this.getChildAt(index)
-                child.apply {
-                    findViewById<ImageView>(R.id.country_list_entry_image).setImageResource(iconRes)
-                    findViewById<TextView>(R.id.country_list_entry_label).text = label
+
+        view = inflate(context, R.layout.view_country_list_entry_flag_container, this)
+        val flagNumberOfColumns = 8
+        var adapterCountryFlags = CountryAdapterFlags(context, _list)
+        context.withStyledAttributes(attributeSet, R.styleable.SimpleStepEntry) {
+
+            flagGrid.layoutManager = GridLayoutManager(context, flagNumberOfColumns)
+            flagGrid.adapter = adapterCountryFlags
+
+            var countryNames = ""
+            val iterator: Iterator<Country> = list!!.iterator()
+            while (iterator.hasNext()) {
+                countryNames += resources.getString(iterator.next().labelRes)
+                if (iterator.hasNext()) {
+                    countryNames += ", "
                 }
             }
+            country_list_entry_label.text = countryNames
+        }
+    }
+}
+
+/**
+ * Adapter for the country flags grid
+ */
+class CountryAdapterFlags internal constructor(
+    context: Context?,
+    countryList: List<Country>?
```

Comment:
Why is the list nullable? I'd prefer it to be not nullable, if there are no countries, the list is empty.

Reply:
it is resolved now